rfcs/042-editable-programmatic-descriptions.md

```diff
@@ -0,0 +1,100 @@
+- Feature Name: editable_programmatic_descriptions
+- Start Date: 2021-07-27
+- RFC PR: [amundsen-io/rfcs#42](https://github.com/amundsen-io/rfcs/pull/42)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000)
+
+# Editable Programmatic Descriptions
+
+## Summary
+
+Programmatic descriptions are a useful way for Amundsen users to store and display business specific metadata about their tables in Amundsen. This feature allows for some customization of Amundsen without customizing the underlying software. Currently, this feature focuses on the machine generated metadata use case. This type of data can be extracted from the source system of truth and ingested into Amundsen via databuilder.
+
+The same need exists for business specific human generated metadata. This type of data does not exist in a machine generated form that can be easily extracted from a source system. Instead, users need to add/edit this metadata directly in the Amundsen UI. 
+
+## Motivation
+
+This feature allows users to configure their Amundsen installation to support one or more editable programmatic descriptions. These editable programmatic descriptions allow users to enrich the metadata displayed on the table details page with additional business specific human generated information. This feature will allow them to add/edit these programmatic descriptions directly in the UI.
+
+## Guide-level Explanation (aka Product Details)
+
+An editable programmatic description is the same as a regular programmatic description, except that it is maintained directly in the UI by a human instead of by a scheduled databuilder job. For example, users could configure an editable programmatic description with a source of data_usage_policy and then update the data_usage_policy on individual table detail pages with a link to the applicable data usage policy on their company's content management site.
+
+Similar to table descriptions, if a table is disabled for editing, then so are the editable programmatic descriptions.
+
+Databuilder jobs may still be useful on a one-off (i.e. non-scheduled basis) to seed Amundsen with manually curated data or perform one-off data conversations.
+
+There will be no impact to existing programmatic descriptions. They will remain non-editable via the UI unless users intentionally update the existing configuration in their Amundsen installation to make them editable.
+
+## UI/UX-level Explanation
+
+Editable programmatic descriptions would be displayed alongside programmatic descriptions in the same way they are displayed today. They will be rendered as markdown, located in one of three sections (left, right, other) in the page and sorted according to display_order.
```

Comment:
afaik currently the order is alphabetical, so this RFC also proposes custom prog description sorting?

Reply:
My proposal would be to keep the sort order the same as it is today. My assumption was that this [function](https://github.com/amundsen-io/amundsen/blob/c4c908df7a14e72058932f4d597656df5a926f48/frontend/amundsen_application/api/utils/metadata_utils.py#L235) sorted the programmatic descriptions of a table according to the `display_order` configured in `PROGRAMMATIC_DISPLAY` (if it was configured). But it sounds like that sorting is not maintained from the API layer to the frontend.

I will update the RFC to call out that sorting will not change.

Reply:
Updated